weblate/trans/views/settings.py

```diff
@@ -198,16 +209,27 @@ def remove(request: AuthenticatedHttpRequest, path):
         messages.success(request, gettext("The translation has been removed."))
     elif isinstance(obj, Component):
         parent = obj.category or obj.project
+        obj.to_delete = True
+        obj.save()
         component_removal.delay(obj.pk, request.user.pk)
         messages.success(
             request, gettext("The translation component was scheduled for removal.")
         )
     elif isinstance(obj, Category):
         parent = obj.category or obj.project
+        categories = [obj]
+        flag_categories_to_delete(categories)
         category_removal.delay(obj.pk, request.user.pk)
         messages.success(request, gettext("The category was scheduled for removal."))
     elif isinstance(obj, Project):
         parent = reverse("home")
+        obj.to_delete = True
+        obj.save()
+        for component in obj.component_set.all():
+            component.to_delete = True
+            component.save()
```

Comment:
Avoid using look when database can do that more effectively:

```suggestion
        obj.component_set.update(to_delete=True)
```

Reply:
Applied this change suggestion.